lib/Target/Lattigo/LattigoEmitter.cpp

```diff
@@ -318,9 +329,88 @@ LogicalResult LattigoEmitter::printOperation(arith::ConstantOp op) {
 }
 
 LogicalResult LattigoEmitter::printOperation(tensor::ExtractOp op) {
-  // only support 1-dim tensor for now
   os << getName(op.getResult()) << " := " << getName(op.getTensor()) << "[";
-  os << getName(op.getIndices()[0]) << "]\n";
+  os << flattenIndexExpression(
+      op.getTensor().getType(), op.getIndices(),
+      [&](Value value) { return variableNames->getNameForValue(value); });
+  os << "]\n";
+  return success();
+}
+
+LogicalResult LattigoEmitter::printOperation(tensor::ExtractSliceOp op) {
+  // Not sure if golang has a strided slice operation like python,
+  // so do it as a loop
+  RankedTensorType resultType = op.getResult().getType();
+
+  if (resultType.getRank() != op.getSourceType().getRank()) {
+    return op.emitError(
+        "Lattigo emitter for ExtractSliceOp only supports "
+        "result rank equal to source rank");
+  }
+
+  // make an array to store the output
+  //
+  //   v := [5][6][7]int32{}
+  //
+  SmallVector<std::string> arrays;
+  for (auto dim : resultType.getShape()) {
+    arrays.push_back("[" + std::to_string(dim) + "]");
+  }
+  std::string resultName = getName(op.getResult(), /*force=*/true);
+  std::string tmpName = resultName + "_array";
+  os << tmpName << " := " << llvm::join(arrays, "")
+     << convertType(resultType.getElementType()) << "{}";
+
+  SmallVector<int64_t> offsets = SmallVector<int64_t>(op.getStaticOffsets());
+  SmallVector<int64_t> sizes = SmallVector<int64_t>(op.getStaticSizes());
+  SmallVector<int64_t> strides = SmallVector<int64_t>(op.getStaticStrides());
```

Comment:
you might need to assert that all the offsets/sizes/strides are static - if they're dynamic then i think these getters will ignore them and you may not assign to all the expected places in the array

Reply:
Done